src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -672,23 +672,10 @@ public override void DidCompleteWithError (NSUrlSession session, NSUrlSessionTas
 
 			void SetResponse (InflightData inflight)
 			{
-				lock (inflight.Lock) {
```

Comment:
We might not need the lock anymore. It's not clear the (only) other usage is needed.

Reply:
I do agree, and I was tempted to remove it, but did not want to add a more complicated change over the one that fixed the leak.